### PR TITLE
Flash Mind Module ( FMM ) Integration

### PR DESCRIPTION
**Pull request by peculiarity**

### CODE REVIEW

File: contracts/multiply/MultiplyProxyActions.sol [hunk 1]
```diff
@@ -785,35 +746,36 @@ contract MultiplyProxyActions {
     );
   }
 
-  function executeOperation(
-    address[] calldata assets,
-    uint256[] calldata amounts,
-    uint256[] calldata premiums,
+  function onFlashLoan(
     address initiator,
```
Comment (adamskrodzki) [inline contracts/multiply/MultiplyProxyActions.sol:751]: this value is supposed to be set to what ? shouldn't we check it to validate right call source?
Or it is users DSProxy?

File: contracts/flashMint/flash.sol [hunk 2]
```diff
@@ -0,0 +1,198 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
```
Comment (adamskrodzki) [inline contracts/flashMint/flash.sol]: Why do we even need that file ?
shouldn't we just use 
IERC3156FlashLender interface instead ?

File: contracts/multiply/MultiplyProxyActions.sol [hunk 3]
```diff
@@ -102,15 +92,11 @@ contract MultiplyProxyActions {
...
-    ILendingPoolV2 lendingPool = getAaveLendingPool(addressRegistry.aaveLendingPoolProvider);
-    lendingPool.flashLoan(
-      addressRegistry.multiplyProxyActions,
-      assets,
-      amounts,
-      modes,
-      address(this),
-      paramsData,
-      AAVE_REFERRAL
+    DssFlash(addressRegistry.lender).flashLoan(
```
Comment (adamskrodzki) [inline contracts/multiply/MultiplyProxyActions.sol]: use IERC3156FlashLender instead

File: contracts/flashMint/base/FlashLoanReceiverBase.sol [hunk 4]
```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
```
Comment (adamskrodzki) [inline contracts/flashMint/base/FlashLoanReceiverBase.sol]: Is this file actually needed ? It is not used anywhere
IERC3156FlashBorrower should be sufficient

File: contracts/flashMint/interface/IVatDaiFlashBorrower.sol [hunk 5]
```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
```
Comment (adamskrodzki) [inline contracts/flashMint/interface/IVatDaiFlashBorrower.sol]: This file is not needed if you use  IERC3156FlashBorrower  instead of implementation

File: contracts/flashMint/interface/IVatDaiFlashLender.sol [hunk 6]
```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
```
Comment (adamskrodzki) [inline contracts/flashMint/interface/IVatDaiFlashLender.sol]: This file is not needed if you reference IERC3156FlashBorrower instead of implementation

**Review by adamskrodzki — CHANGES_REQUESTED** [review]
Diff can be much smaller

File: contracts/multiply/MultiplyProxyActions.sol (hunk 1 above)
Comment (peculiarity) [reply contracts/multiply/MultiplyProxyActions.sol:751]: I know that it could be checked. I've seen the example in the doc as well. It's the DsPRoxy address.  There is even the comment taht I left there right ? So yeah, it's the proxyAddress and In order to get that, we have to pass that within the AddressRegistry .. which I think we can skip.

Comment (adamskrodzki) [reply contracts/multiply/MultiplyProxyActions.sol:751]: I guess we actually have all that's needed - there is cdpManager in AddressRegistry there is cdpId in CdpData
so we can check if initiator is allowed to operate the vault. But i agree that it is more "Nice to have" than "must have"

**Review by adamskrodzki — APPROVED** [review]

**Review by fenris85 — APPROVED** [review]
